src/storage.rs

```diff
@@ -52,8 +54,31 @@ pub fn address_to_key(prefix: KeyPrefix, address: &Address) -> [u8; 22] {
     result
 }
 
+pub enum StorageKeyKind {
+    Normal([u8; 54]),
+    Generation([u8; 58]),
+}
+
+impl AsRef<[u8]> for StorageKeyKind {
+    fn as_ref(&self) -> &[u8] {
+        use StorageKeyKind::*;
+        match self {
+            Normal(v) => v.as_slice(),
+            Generation(v) => v.as_slice(),
+        }
+    }
+}
+
+pub fn storage_to_key(address: &Address, key: &H256, generation: u32) -> StorageKeyKind {
+    if generation == 0 {
+        StorageKeyKind::Normal(normal_storage_key(address, key))
+    } else {
+        StorageKeyKind::Generation(generation_storage_key(address, key, generation))
+    }
+}
+
 #[allow(dead_code)]
-pub fn storage_to_key(address: &Address, key: &H256) -> [u8; 54] {
+pub fn normal_storage_key(address: &Address, key: &H256) -> [u8; 54] {
```

Comment:
Remove `pub` we should only be using `storage_to_key`

Reply:
Good point, as there are all internal.